features/features.go

```diff
@@ -224,6 +224,13 @@ var (
 					enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
 					mustRegister()
 
+	FeatureGateAdvancedRoutingBGP = newFeatureGate("AdvancedRoutingBGP").
```

Comment:
Since the feature is (from what I can tell) talking mostly about FRR, should this be called BGP? (I don't even know what these acronyms are so, are they different?)

Reply:
So FRR is a well known thing and a name in itself: https://frrouting.org/
In its k8s form: https://github.com/metallb/frr-k8s
Which is what will be deployed from CNO operator when FRR is set as provider

It's basically a suite of linux daemons providing support for multiple routing protocols and related functionality. BGP is just one of such protocols and the main use case we want to cover.

I don't know what's the relevance of choosing one name vs any other name for the feature gate itself, but we can go for `AdvancedRoutingCNO` or `AdvancedRoutingFRR` if that makes more sense to you. They are all fine to me.

Reply:
Rather than FFR, would it be more obvious if we make the enum value `FRRouting`? That appears to be the full name of this if I skimmed the site linked properly.

If we are renaming the field to include capabilities, would it make sense to include that in the feature gate name as well?

Reply:
I'm also wondering if focusing on the FRRouting name is not what we want here. Is that not an implementation detail? Should we not be trying to name this based on the end user capability that they gain from enabling this?

Reply:
> I'm also wondering if focusing on the FRRouting name is not what we want here. Is that not an implementation detail? Should we not be trying to name this based on the end user capability that they gain from enabling this?

I touched on it in a previous comment: https://github.com/openshift/api/pull/1944#discussion_r1672512492
The capabilities are acquired via enabling additional components or providers. Then those components/providers have their own APIs outside of the OCP API to configure exactly those capabilities. There might be exceptions but that is the general approach.

> Rather than FFR, would it be more obvious if we make the enum value `FRRouting`? That appears to be the full name of this if I skimmed the site linked properly.
> 
> If we are renaming the field to include capabilities, would it make sense to include that in the feature gate name as well?

Consider `FRR` vs `FRRouting` like `go` vs `golang`, both forms are well known to refer to it. However, we can break out of the `FRR` vs `FRRouting` discussion and actually go with `FRR-k8s` which is the actual name of what we are going to deploy.

We can imagine in the future the possibility there is `something like FRR-k8s but different that also deploys FRR and provides FRR capabilities in a different way or something` that we want to add as provider as well. The we would have a mutually exclusive validation if needed.